experimental/LieAlgebras/src/PathModel.jl

```diff
@@ -0,0 +1,617 @@
+##### Kashiwara Crystal #####
+
+abstract type AbstractCrystal <: AbstractAlgebra.Set end
+
+abstract type AbstractCrystalElem <: AbstractAlgebra.SetElem end
+
+# AbstractCrystalElem required methods
+
+function Base.eps(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+function phi(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+@doc raw"""
+    Apply the root operator $\tilde e_i$ to `b` in place and return the result.
+"""
+function ealpha!(b::AbstractCrystalElem, i::Int)
+  error("not implemented")
+end
+
+@doc raw"""
+    Apply the root operator $\tilde f_i$ to `b` in place and return the result.
+"""
+function falpha!(b::AbstractCrystalElem, i::Int)
+  error("not implemented")
+end
+
+@doc raw"""
+    Return the weight of `b`.
+"""
+function weight(b::AbstractCrystalElem)
+  error("not implemented")
+end
+
+# AbstractCrystal optional methods
+
+@doc raw"""
+    is_upper_normal(B::AbstractCrystal) -> bool
```

Comment:
```suggestion
    is_upper_normal(B::AbstractCrystal) -> Bool
```
types in julia are uppercase (more similar places below)